From 2c55ccdaef8901ee4c6f8a8feda77dfc41f0347b Mon Sep 17 00:00:00 2001
From: Haihao Xiang <haihao.xiang@intel.com>
Date: Wed, 12 Feb 2020 16:38:45 +0800
Subject: [PATCH] Support up to 8Kx8K VP9 encoding on GEN12

Otherwise the driver only advertises 1920x1920 VP9 encoding on GEN12

Refer to
https://gitlab.freedesktop.org/gstreamer/gstreamer-vaapi/issues/229 for
issue details

Signed-off-by: Haihao Xiang <haihao.xiang@intel.com>
---
 media_driver/linux/gen12/ddi/media_libva_caps_g12.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/media_driver/linux/gen12/ddi/media_libva_caps_g12.cpp b/media_driver/linux/gen12/ddi/media_libva_caps_g12.cpp
index 26c5fb664f73..830e62da7df2 100644
--- a/media_driver/linux/gen12/ddi/media_libva_caps_g12.cpp
+++ b/media_driver/linux/gen12/ddi/media_libva_caps_g12.cpp
@@ -1096,6 +1096,10 @@ VAStatus MediaLibvaCapsG12::QuerySurfaceAttributes(
         {
             attribs[i].value.value.i = CODEC_16K_MAX_PIC_WIDTH;
         }
+        else if(IsVp9Profile(profile))
+        {
+            attribs[i].value.value.i = CODEC_8K_MAX_PIC_WIDTH;
+        }
         if(IsAvcProfile(profile))
         {
             attribs[i].value.value.i = CODEC_4K_MAX_PIC_WIDTH;
@@ -1114,6 +1118,10 @@ VAStatus MediaLibvaCapsG12::QuerySurfaceAttributes(
         {
             attribs[i].value.value.i = CODEC_16K_MAX_PIC_HEIGHT;
         }
+        else if(IsVp9Profile(profile))
+        {
+            attribs[i].value.value.i = CODEC_8K_MAX_PIC_HEIGHT;
+        }
         if(IsAvcProfile(profile))
         {
             attribs[i].value.value.i = CODEC_4K_MAX_PIC_HEIGHT;

